docs: form component changes (#216)
centrica-engineering/nucleus-docs: docs/components/ns-form.md

## Introduction

> The form component is used to collect customer input and provide personal experience to customer based on their needs.

> The `ns-form` component mimics the native form element features. It's main purpose is to collect customer input and validate the input before sending them to processing.

Reviewer: Minor tweak.

```suggestion
The `ns-form` component mimics the native form element. It's main purpose is to collect customer data and validate the input before sending the data for processing.
```
